src/providers/vmware/vmware.go

```diff
@@ -43,3 +48,46 @@ func (p provider) ShouldRetry() bool {
 func (p *provider) BackoffDuration() time.Duration {
 	return 0
 }
+
+func decodeData(data string, encoding string) ([]byte, error) {
+	switch encoding {
+	case "":
+		return []byte(data), nil
+
+	case "b64", "base64":
+		return decodeBase64Data(data)
+
+	case "gz", "gzip":
+		return decodeGzipData(data)
+
+	case "gz+base64", "gzip+base64", "gz+b64", "gzip+b64":
```

Comment:
We have to support the same options as cloudinit: https://github.com/coreos/coreos-cloudinit/blob/master/config/decode.go#L34. I'm not going to document the alternate names, because I don't want to support them long term.